src/icons/AssetIcon.tsx

```diff
@@ -18,20 +19,19 @@ interface KindIconProps {
 export const KindIcon = (props: KindIconProps) => {
     const size = props.size || 16;
     const theme = useTheme().palette.mode;
-    const isDarkMode = theme === 'dark';
     const style = {
         fontSize: size + 'px',
         height: size + 'px',
     };
+    const icons = props.icons ?? ([props.icon].filter(Boolean) as IconValue[]);
```

Comment:
Moved the icon/icons logic into the innermost KindIcon instead of the two wrappers.